### ISSUE TITLE
Implement AEAD command line tool

### ISSUE THREAD

**Issue by paulidale** [openssl/project#248]

**Comment by beldmit** [comment]
I'd say it's worth having an explicit command line key allowing seeing what was decrypted despite tag calculation failure.

**Comment by paulidale** [comment]
I've a healthy skepticism about doing anything with a non-authenticated decryption of an AEAD mode cipher.  It feels like it defeats the purpose.

I would accept a `-stream` option that blocks the data in manageable sized authenticated lumps and letting them be emitted in the face of some failures to authenticate.  That's the original use case.

The default for such a setting doesn't much concern me.  I.e. should `-stream` or `-no-stream` (or `-batch`?) be the default?


I'd also like the metadata to contain enough information to decrypt without knowing any details (beyond the key).


Should we also do a utility command for XTS mode?  That has been asked for in the past and is broadly similar.

**Comment by t8m** [comment]
Yeah. Please note that designing the file format to be future-proof is fairly non-trivial.  For example the size of the separately encrypted/authenticated blocks should be variable. In the case of encrypted video content the authenticated blocks should be cut at the end of a video frame data so if any block is malformed it affects the least number of video frames as possible.

For aes-xts - should the format be compatible with dm-crypt and the metadata with cryptsetup-luks?
